laws/src/test/scala/cats/effect/Generators.scala

```diff
@@ -202,6 +233,48 @@ trait ConcurrentGenerators[F[_], E] extends ApplicativeErrorGenerators[F, E] {
     } yield back
 }
 
+trait TemporalGenerators[F[_], E] extends ConcurrentGenerators[F, E] with ClockGenerators[F] {
+  implicit val F: Temporal[F, E]
+
+  override protected def baseGen[A: Arbitrary: Cogen] =
+    ("sleep" -> genSleep[A]) :: super.baseGen[A]
+
+  private def genSleep[A: Arbitrary] =
+    for {
+      t <- arbitraryFD.arbitrary
+      a <- arbitrary[A]
+    } yield F.sleep(t).as(a)
+}
+
+trait AsyncGenerators[F[_]] extends TemporalGenerators[F, Throwable] with SyncGenerators[F] {
+  implicit val F: Async[F]
+
+  protected implicit val arbitraryEC: Arbitrary[ExecutionContext]
+  protected implicit val cogenFU: Cogen[F[Unit]]
+
+  override protected def recursiveGen[A: Arbitrary: Cogen](deeper: GenK[F]) =
+    List(
+      "async" -> genAsync[A](deeper),
+      "evalOn" -> genEvalOn[A](deeper)) ++ super.recursiveGen[A](deeper)
+
+  private def genAsync[A: Arbitrary: Cogen](deeper: GenK[F]) =
+    for {
+      result <- arbitrary[Either[Throwable, A]]
+
+      fo <- deeper[Option[F[Unit]]](
+        Arbitrary(
+          Gen.option[F[Unit]](
+            deeper[Unit](Arbitrary(arbitrary[Unit]), Cogen.cogenUnit))),
```

Comment:
```suggestion
            deeper[Unit],
```

please tell me this just works ;)

Reply:
I don't think it does! :-( I'll check though